src/bindings/python/flux/accounting/job_archive_interface.py

```diff
@@ -288,6 +311,77 @@ def calc_usage_factor(
 
     usage_user_current = sum(per_job_factors)
 
+    # if no new jobs were found and we are still in the same half-life
+    # period, then the current usage remains the same
+    if len(job_totals_user) == 0 and (
+        float(t_end_half_life_period) > (time.time() - half_life_period)
+    ):
+        # fetch past usage factors
+        past_usage_factors = fetch_old_usage_factors(acct_conn, user, bank)
+
+        usage_user_historical = sum(past_usage_factors)
+    # if no new jobs were found but we are past the most recent half-life
+    # period, then the current usage needs to have a decay factor applied
+    # to it
+    elif len(job_totals_user) == 0 and (
+        float(t_end_half_life_period) < (time.time() - half_life_period)
+    ):
+        # fetch past usage factors
+        past_usage_factors = fetch_old_usage_factors(acct_conn, user, bank)
+
+        usage_user_historical = apply_decay_factor(0.5, acct_conn, user, bank)
+
+        # update new end of half-life period timestamp
+        update_timestamp_stmt = """
+            UPDATE t_half_life_period_table
+            SET end_half_life_period=?
+            WHERE cluster='cluster'
+            """
+        acct_conn.execute(
+            update_timestamp_stmt, ((float(t_end_half_life_period) + half_life_period),)
+        )
```

Comment:
Because this value applies to the entire table, this update should only happen after every user/bank has it's new period and decay factors applied.

It would work here if it only applied to an individual bank/user row.

Reply:
Thanks for pointing this out. Maybe then the update _should_ occur per user/bank combination, instead of globally? Maybe then, the `end_half_life_period` column could instead just be added to the `job_usage_factor_table` for every user/bank combination, instead of in its own table?

Reply:
Or, maybe this update can happen after every user/bank has its new period and decay factors applied. Maybe the cron job that calls `calc_usage_factor()` can have a workflow like the following, where:

- it iterates through every user/bank combination and calls `calc_usage_factor()` for each one

- once it finishes and every user/bank combo is updated, it can check to see if we are in still in the same half-period, and if not, update `t_half_life_period_table` with the new `end_half_life_period` timestamp. I think that could remove the `update_timestamp_stmt` from this `if-else` branch. 

Does that make sense?

Reply:
Yeah. I think either of those would work.  I have a slight preference for the latter, since the period end time conceptually seems like it should be independent of the user/bank, and it might be a little weird for every user/bank to have their own end periods.

Reply:
I also have a preference for the latter - I think it makes sense to keep it separate from updating individual user/bank values. `calc_usage_factor()` was getting pretty long as it is and it would probably be cleaner to move it into its own function. That implementation should be pushed up with the new `fixup` commits.